python_modules/dagster/dagster/_daemon/monitoring/monitoring_daemon.py

```diff
@@ -3,35 +3,43 @@
 import time
 from typing import Iterator, Optional
 
+import pendulum
+
 from dagster import (
     DagsterInstance,
     _check as check,
 )
-from dagster._core.events import DagsterEventType
+from dagster._core.events import DagsterEventType, EngineEventData
 from dagster._core.launcher import WorkerStatus
 from dagster._core.storage.pipeline_run import (
     IN_PROGRESS_RUN_STATUSES,
-    DagsterRun,
     DagsterRunStatus,
+    RunRecord,
     RunsFilter,
 )
+from dagster._core.storage.tags import MAX_RUNTIME_TAG
 from dagster._core.workspace.context import IWorkspace, IWorkspaceProcessContext
 from dagster._utils import DebugCrashFlags
 from dagster._utils.error import SerializableErrorInfo, serializable_error_info_from_exc_info
 
+DEFAULT_MAX_RUNTIME = 60 * 60 * 12
 RESUME_RUN_LOG_MESSAGE = "Launching a new run worker to resume run"
 
 
 def monitor_starting_run(
-    instance: DagsterInstance, run: DagsterRun, logger: logging.Logger
+    instance: DagsterInstance, run_record: RunRecord, logger: logging.Logger
 ) -> None:
+    run = run_record.dagster_run
     check.invariant(run.status == DagsterRunStatus.STARTING)
     run_stats = instance.get_run_stats(run.run_id)
 
     launch_time = check.not_none(
         run_stats.launch_time, "Run in status STARTING doesn't have a launch time."
     )
-    if time.time() - launch_time >= instance.run_monitoring_start_timeout_seconds:
+    if (
+        instance.run_launcher.supports_check_run_worker_health
```

Comment:
this check isn't needed is is?

Reply:
I believe it is needed - since the run termination stuff isn't bound by the same constraints as run worker health checks, I got rid of the check for having the run monitoring daemon only enabled when certain run launchers are usable, and instead moved the check here.

Reply:
That's to guard the calls to `check_run_worker_health`, which we aren't calling here. That may change soon with @gibsondan 's debug improvements but then we could just gate that

There's no particular reason why starting timeouts weren't supported for all run launchers so if we're making the change for max run time then may as well support it for starting too